yahooquery/utils/__init__.py

```diff
@@ -69,31 +161,29 @@ def _init_session(session=None, **kwargs):
                 max_retries=retries, timeout=kwargs.get("timeout", DEFAULT_TIMEOUT)
             ),
         )
-        # TODO: Figure out how to utilize this within the validate_response
-        # TODO: This will be a much better way of handling bad requests than
-        # TODO: what I'm currently doing.
-        # session.hooks['response'] = \
-        #     [lambda response, *args, **kwargs: response.raise_for_status()]
-        user_agent = kwargs.get("user_agent", random.choice(USER_AGENT_LIST))
-        session_headers["User-Agent"] = user_agent
-        if kwargs.get("headers") and isinstance(kwargs.get("headers"), dict):
-            session_headers.update(**headers)
-        session.headers.update(**session_headers)
-    return session
-
-
-# def get_cookies(user_agent):
-#     options = webdriver.ChromeOptions()
-#     options.add_argument('--user-agent=' + user_agent)
-#     options.add_argument('headless')
-#     driver = webdriver.Chrome(
-#         ChromeDriverManager().install(), chrome_options=options)
-#     driver.get("https://finance.yahoo.com/screener/new")
-#     cookies = driver.get_cookies()
-#     driver.quit()
-#     return cookies
+        session, crumb = setup_session_with_cookies_and_crumb(session)
+    return session, crumb
 
 
+def setup_session_with_cookies_and_crumb(session: Session):
+    headers = {**random.choice(HEADERS), **addl_headers}
+    session.headers = headers
+    response = session.get('https://finance.yahoo.com', hooks={'response': get_crumb})
```

Comment:
@ValueRaider This is the place where you would modify to (I'm guessing) `https://uk.finance.yahoo.com`

Reply:
I wonder if it has something to do with accepting their cookies. I tried navigating to the uk site and got their consent screen.

![image](https://github.com/dpguthrie/yahooquery/assets/10198967/502b8118-fd77-493e-8ff4-e5e594e82548)

I’m guessing without accepting that then YF is unable to actually set the appropriate cookies.

Reply:
Also, this code doesn’t use selenium. It’s only used to login to YF (and not even sure that’s working right now as it looks like they’ve added a captcha)

Reply:
The getcrumb `response.text` contains string "HTTP Status 403 - Forbidden". Is that typical of Yahoo expecting a cookie?

Reply:
Sorry wrong part of code, that 403 happens in `setup_session_with_cookies_and_crumb()` on `https://finance.yahoo.com`

Reply:
That's really surprising!  I'm not sure what would make that a forbidden request just to the YF home page.